Universal/Sniffs/CodeAnalysis/MixedBooleanOperatorSniff.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * PHPCSExtra, a collection of sniffs and standards for use with PHP_CodeSniffer.
+ *
+ * @package   PHPCSExtra
+ * @copyright 2021 WoltLab GmbH, 2023 PHPCSExtra Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCSStandards/PHPCSExtra
+ */
+
+namespace PHPCSExtra\Universal\Sniffs\CodeAnalysis;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Util\Tokens;
+use PHPCSUtils\BackCompat\BCFile;
+
+/**
+ * Forbid mixing different binary boolean operators within a single expression without making precedence
+ * clear using parentheses.
+ *
+ * @link https://github.com/squizlabs/PHP_CodeSniffer/pull/3205
+ * @link https://github.com/php-fig/per-coding-style/issues/22
+ *
+ * @since 1.2.0
+ */
+final class MixedBooleanOperatorSniff implements Sniff
+{
+    private $previousTokens = [];
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @since 1.2.0
+     *
+     * @return array<int|string>
+     */
+    public function register()
+    {
+        $this->previousTokens = \array_merge(
+            Tokens::$booleanOperators,
+            [\T_INLINE_THEN, \T_INLINE_ELSE]
+        );
+
+        return Tokens::$booleanOperators;
+    }
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @since 1.2.0
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $start = BCFile::findStartOfStatement($phpcsFile, $stackPtr);
+
+        $previous = $phpcsFile->findPrevious(
+            $this->previousTokens,
+            $stackPtr - 1,
+            $start,
+            false,
+            null,
+            true
+        );
+
+        if (
+            // No token found.
+            $previous === false
+            // Identical operator found.
+            || $tokens[$previous]['code'] === $tokens[$stackPtr]['code']
+            // Beginning of the expression found for the ternary conditional operator.
+            || \in_array($tokens[$previous]['code'], [\T_INLINE_THEN, \T_INLINE_ELSE], true)
```

Comment:
Might just be me, but why use `in_array()` when a plain comparison will do ? It's not as if we need to search a huge array.
```suggestion
            || $tokens[$previous]['code'] === \T_INLINE_THEN
            || $tokens[$previous]['code'] === \T_INLINE_ELSE
```

Reply:
Using `\in_array()` keeps related logic within a single expression.

Reply:
Please resolve this if no change is requested / if my response is sufficiently satisfying.

Reply:
I'm fine leaving this as-is for now. I'll switch it to an `isset()` once the `Collections::ternaryOperators()` method is available in PHPCSUtils.